### DISCUSSION TITLE
CustomUser

### DISCUSSION THREAD

**Discussion by lrnzdc3696** [encode/django-rest-framework#8416] [Ideas & Suggestions]
Context:
I am working on a route where a person can make an account. I have made a custom user because I want them to be able to make an account via email and not username.

Problem:
The code is not "dynamic" because the username is hardcoded.
In here the username is hard coded.
<https://github.com/encode/django-rest-framework/blob/070c32f4a62ef0544f58de404c87d86db36fd825/rest_framework/authtoken/serializers.py#L8>

I don't want to use a third party lib.
I don't want to do some hacky stuff.

Solution in mind:
maybe use the get_user_model?

**Comment by foarsitter** [comment]
But do you want to override the [ObtainAuthToken(APIView)](https://github.com/encode/django-rest-framework/blob/71e6c30034a1dd35a39ca74f86c371713e762c79/rest_framework/authtoken/views.py#L11) class with an custom serializer class?

**Reply by lrnzdc3696** [reply]
I already have

**Reply by foarsitter** [reply]
Well, then what is the problemen?

**Reply by lrnzdc3696** [reply]
It would be much nicer if I don't have to subclass and overwrite stuff.

**Comment by hashlash** [comment]
Can't you just pass the user's email as `username`?

And just to make sure, have you set your custom user model's [`USERNAME_FIELD`](https://docs.djangoproject.com/en/4.0/topics/auth/customizing/#django.contrib.auth.models.CustomUser.USERNAME_FIELD) to your mail field?

**Reply by lrnzdc3696** [reply]
Well I can, but people might get confused about it.

**Reply by hashlash** [reply]
I don't have any particular stand on this.

You could tell your front-end developers if you are building a private API. Or, if it's for public API, you could give an help_text on the docs.

But if you insist that this must be a feature, I think it's more proper to be on the **Ideas & Suggestions** category. :slightly_smiling_face:

**Reply by lrnzdc3696** [reply]
alright thank you
